integration-tests/tests/link-to-new-workspace.js

```diff
@@ -21,6 +16,7 @@ const testLinkToNewWorkspaceFn = withUserToken(async ({ testUrl, page, token })
     // Wait for bucket access to avoid sporadic failures
     await checkBucketAccess(page, newWorkspaceBillingAccount, newWorkspaceName)
     await findText(page, `${newWorkspaceBillingAccount}/${newWorkspaceName}`)
+    await findText(page, 'Select a data type')
   } finally {
     try {
       await page.evaluate((name, billingProject) => {
```

Comment:
Can't you just call `deleteWorkspace` here?

Reply:
If for some reason you cannot, you probably want to throw an error if the workspace can't be deleted, like you did in `deleteWorkspace`.

Reply:
If the test fails to create a workspace, this delete throws an error. If it's not handled that error obscures the original test error and makes it difficult to debug. Although now that I think of it I could check if the workspace exists before calling `deleteWorkspace` to avoid that situation.

Reply:
If a workspace isn't created, won't the test fail earlier? checkBucketAccess will error I believe, and you shouldn't be able to navigate to the workspace page.

Reply:
The finally block runs even if there's an error thrown in the try block, so the original error gets lost

Reply:
This comment from Pete has a useful code snippet to help observe this behavior: https://github.com/DataBiosphere/terra-ui/pull/2807#pullrequestreview-875241605

Reply:
I still think you can call `deleteWorkspace` and just catch any errors that it throws. That way you would not duplicate the deleteWorkspace code.

Reply:
@cahrens Are you referring to [this deleteWorkspace](https://github.com/DataBiosphere/terra-ui/blob/bf9423b28bcf8d0098c076e61e8dd28f5b7f62f3/integration-tests/utils/integration-helpers.js#L42) function?

If so, I tried out this function. It doesn't look like a super-clean abstraction and ends up running some additional code around sign in that I don't think we want to do. We could add a ticket to investigate creating a cleaner utility for this.

Reply:
OK, I was hoping it would be straightforward to use, but if not feel free to punt it.

Reply:
@petesantos and I paired on improving the error handling here! Now if the test fails it will throw the test failure (and log the cleanup error if that fails too). If the test succeeds and the cleanup fails, it throws the cleanup error.